warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/handler/shard/ShardedDataTypeHandlerTest.java

```diff
@@ -1,17 +1,176 @@
+import datawave.data.type.NumberType;
+import datawave.ingest.config.RawRecordContainerImpl;
+import datawave.ingest.data.RawRecordContainer;
+import datawave.ingest.data.Type;
+import datawave.ingest.data.TypeRegistry;
+import datawave.ingest.data.config.DataTypeHelper;
+import datawave.ingest.data.config.MaskedFieldHelper;
+import datawave.ingest.data.config.NormalizedContentInterface;
+import datawave.ingest.data.config.NormalizedFieldAndValue;
+import datawave.ingest.data.config.ingest.BaseIngestHelper;
+import datawave.ingest.data.config.ingest.ContentBaseIngestHelper;
+import datawave.ingest.mapreduce.job.BulkIngestKey;
+import datawave.ingest.mapreduce.partition.BalancedShardPartitioner;
+import datawave.ingest.protobuf.Uid;
+import datawave.ingest.table.config.ShardTableConfigHelper;
+import datawave.ingest.table.config.TableConfigHelper;
+import datawave.policy.IngestPolicyEnforcer;
+import datawave.util.TableName;
+import org.apache.accumulo.core.data.Value;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.io.Text;
 import org.apache.hadoop.mapreduce.TaskAttemptID;
 import org.apache.hadoop.mapreduce.task.TaskAttemptContextImpl;
+import org.junit.Before;
 import org.junit.Test;
 
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.Assert.assertTrue;
+
 public class ShardedDataTypeHandlerTest {
     
-    @Test(expected = IllegalArgumentException.class)
-    public void testSetup() {
-        Configuration conf = new Configuration();
-        ShardedDataTypeHandler<Text> handler = new AbstractColumnBasedHandler<>();
-        handler.setup(new TaskAttemptContextImpl(conf, new TaskAttemptID()));
+    ShardedDataTypeHandler<Text> handler;
+    AbstractColumnBasedHandler<Text> dataTypeHandler;
+    TestIngestHelper ingestHelper;
+    TestMaskedHelper maskedFieldHelper;
+    private static final int NUM_SHARDS = 241;
+    private static final String DATA_TYPE_NAME = "wkt";
+    private static final String WKT_BYTE_LENGTH_FIELD = "WKT_BYTE_LENGTH";
+    private static final String GEO_FIELD = "GEO";
+    private static final String INGEST_HELPER_CLASS = TestIngestHelper.class.getName();
+    
+    Configuration configuration;
+    
+    public static class TestIngestHelper extends ContentBaseIngestHelper {
+        @Override
+        public Multimap<String,NormalizedContentInterface> getEventFields(RawRecordContainer record) {
+            Multimap<String,NormalizedContentInterface> eventFields = HashMultimap.create();
+            String rawRecord = new String(record.getRawData());
+            for (String value : rawRecord.split(";")) {
+                NormalizedContentInterface geo_nci = new NormalizedFieldAndValue("FIELD_NAME", value);
+                eventFields.put("FIELD_NAME", geo_nci);
+            }
+            return normalizeMap(eventFields);
+        }
+        
+        public String getNormalizedMaskedValue(final String key) {
+            return "FIELD_NAME";
```

Comment:
Might be worth making this `MASKED_VALUE` or something like that. Your unit test below asserts that "FIELD_NAME" reversed is returned. But that had me scratching my head wondering "Why are we reversing the FIELD_NAME?". It just so happens you pass in FIELD_NAME to the createTermIndexColumn() in your test too.

If you change this to MASKED_VALUE it'll make it more obvious that this is a dummy version of a masked value an different from the field name.

Reply:
Updated in lastest commit